opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py

```diff
@@ -46,6 +46,8 @@ def http_status_to_status_code(
         status (int): HTTP status code
     """
     # See: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status
+    if status is None:
+        return StatusCode.UNSET
```

Comment:
Not sure about encoding knowledge about a special case in this function. IMO we should either have a `isinstance(status, int)` check or wrap in a try/except block to cover all cases of invalid input

Reply:
Agreed.

Would it also make sense to also try to coerce status to an integer in case the status was passed as a string representation while we're looking at this?

Reply:
Not sure about that. We don't necessarily want to work with a wide range of inputs. We just don't ever want to crash. I think the caller should be responsible for converting types.

Reply:
Is this something common with other http client libraries as well? I personally prefer not to have the input sanitisation part here. And the caller would have the more context when the status is unusual and it can decide to what to do instead of we returning the UNSET here for all such scenarios.

Reply:
Perhaps we should just include the check only for this specific instrumentation.